documentation/examples/rbac-setup.yml

```diff
@@ -32,6 +32,17 @@ kind: ServiceAccount
 metadata:
   name: prometheus
   namespace: default
+secrets:
+  - name: prometheus-sa-token
```

Comment:
this list is for referencing secrets that are to be mounted into pods... don't include manually created token secrets in this list

Reply:
No, creation of the Secret object is fine (matches the method described in https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-long-lived-api-token-for-a-serviceaccount)

addition of the name of that manually created Secret to the `.secrets` list field in the ServiceAccount API object is the part that isn't correct. 

Doing that makes the manually created Secret appear to be an autogenerated secret, and encourages/enables locating the secret via that field, which is not what that field is for.

Reply:
I added this because if the user apply "kubernetes.io/enforce-mountable-secrets" annotation, pod is only allowed to use secrets which are mentioned in this list.

According to K8s API docs:
Pods are only limited to this list if this service account has a "kubernetes.io/enforce-mountable-secrets" annotation set to "true".

We can drop the list and just relying on "kubernetes.io/service-account.name: prometheus" in Secret.

Reply:
I don't see evidence that this secret is being mounted into the pod, so inclusion in the list should not be required, right? Secret-based tokens have not been auto-mounted into pods since 1.21

Reply:
Personally, I'm doing that for future isolation, but right now we can drop it and make user able for future decisions.